From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GliczDev <67753196+GliczDev@users.noreply.github.com>
Date: Mon, 26 Aug 2024 15:13:50 +0200
Subject: [PATCH] Entity API


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 9905ba66541f828d12f0497934eb8101b4c4c53a..43055cd7a9ef275af8c00c0a65625a0e51d7ff08 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -242,6 +242,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     private boolean hasVisualFire;
     @Nullable
     private BlockState inBlockState = null;
+    private final me.glicz.airflow.entity.AirEntity airEntity; // Airflow - Entity API
 
     public Entity(EntityType<?> entityType, Level level) {
         this.type = entityType;
@@ -263,8 +264,19 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
         this.entityData = builder.build();
         this.setPos(0.0, 0.0, 0.0);
         this.eyeHeight = this.dimensions.eyeHeight();
+        this.airEntity = createAirEntity(); // Airflow - Entity API
     }
 
+    // Airflow start - Entity API
+    protected me.glicz.airflow.entity.AirEntity createAirEntity() {
+        return new me.glicz.airflow.entity.AirEntity(this);
+    }
+
+    public me.glicz.airflow.entity.AirEntity getAirEntity() {
+        return this.airEntity;
+    }
+    // Airflow end - Entity API
+
     public boolean isColliding(BlockPos pos, BlockState state) {
         VoxelShape collisionShape = state.getCollisionShape(this.level(), pos, CollisionContext.of(this));
         VoxelShape voxelShape = collisionShape.move(pos.getX(), pos.getY(), pos.getZ());
diff --git a/net/minecraft/world/entity/EntityType.java b/net/minecraft/world/entity/EntityType.java
index 7c1ee0965de421a195fba2f2bba953dc695f94ee..798206e1a70fa98f3a59cf9aa721977cf85e9c54 100644
--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -833,6 +833,7 @@ public class EntityType<T extends Entity> implements FeatureElement, EntityTypeT
     private final EntityDimensions dimensions;
     private final float spawnDimensionsScale;
     private final FeatureFlagSet requiredFeatures;
+    public final me.glicz.airflow.entity.AirEntityType airEntityType = new me.glicz.airflow.entity.AirEntityType(this); // Airflow - Entity API
 
     private static <T extends Entity> EntityType<T> register(String key, EntityType.Builder<T> builder) {
         return Registry.register(BuiltInRegistries.ENTITY_TYPE, key, builder.build(key));
