From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GliczDev <67753196+GliczDev@users.noreply.github.com>
Date: Wed, 28 Aug 2024 21:03:48 +0200
Subject: [PATCH] Item API


diff --git a/net/minecraft/world/item/Item.java b/net/minecraft/world/item/Item.java
index ed7d2e578643f422f11d5354c8be4e2374686480..deb369552452a19d372d30134ee189153ba119ea 100644
--- a/net/minecraft/world/item/Item.java
+++ b/net/minecraft/world/item/Item.java
@@ -71,6 +71,7 @@ public class Item implements FeatureElement, ItemLike {
     @Nullable
     private String descriptionId;
     private final FeatureFlagSet requiredFeatures;
+    public final me.glicz.airflow.item.AirItemType airItemType = new me.glicz.airflow.item.AirItemType(this); // Airflow - Item API
 
     public static int getId(Item item) {
         return item == null ? 0 : BuiltInRegistries.ITEM.getId(item);
diff --git a/net/minecraft/world/item/ItemStack.java b/net/minecraft/world/item/ItemStack.java
index 8224c249c52cdba264acdd85cd07e3163140f4fd..014099568e69e8af1466645ac0fedd74de9778c9 100644
--- a/net/minecraft/world/item/ItemStack.java
+++ b/net/minecraft/world/item/ItemStack.java
@@ -181,6 +181,7 @@ public final class ItemStack implements DataComponentHolder {
     final PatchedDataComponentMap components;
     @Nullable
     private Entity entityRepresentation;
+    public final me.glicz.airflow.item.stack.AirItemStack airItemStack = new me.glicz.airflow.item.stack.AirItemStack(this); // Airflow - Item API
 
     private static DataResult<ItemStack> validateStrict(ItemStack stack) {
         DataResult<Unit> dataResult = validateComponents(stack.getComponents());
