From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GliczDev <67753196+GliczDev@users.noreply.github.com>
Date: Fri, 30 Aug 2024 20:51:42 +0200
Subject: [PATCH] World API


diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index 4150c00a9b5deff99cd30c1e70c9436b8f9a8039..d1832a31dbc257e297b4d9af15d5f673dc79ac01 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -207,6 +207,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
     private final StructureCheck structureCheck;
     private final boolean tickTime;
     private final RandomSequences randomSequences;
+    public final me.glicz.airflow.world.AirWorld airWorld = new me.glicz.airflow.world.AirWorld(this); // Airflow - World API
 
     public ServerLevel(
         MinecraftServer server,
